src/DataLib/fendl32B_retrofit/groupr_tools.py

```diff
@@ -110,16 +112,57 @@ def write_njoy_input_file(template):
     with open('groupr.inp', 'w') as f:
         f.write(template)
 
+def insert_after_identifier(file_str, identifier, new_line):
+    """
+    Insert a new line immediately after the last occurrence of a line
+    containing the specified identifier in the file string.
+
+    Arguments:
+        file_str (str): The content of the GENDF file as a string.
+        identifier (str): The identifier to search for in the file content.
+        new_line (str): The new line to insert after the identified line.
```

Comment:
might be a better variable name than `new_line`.  Perhaps `record_for_insertion` or something since `new_line` means something specific